servicetalk-transport-api/src/main/java/io/servicetalk/transport/api/ConnectionObserver.java

```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright © 2020 Apple Inc. and the ServiceTalk project authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.servicetalk.transport.api;
+
+/**
+ * An observer interface that provides visibility into events associated with a single network connection.
+ */
+public interface ConnectionObserver {
+
+    /**
+     * Notifies when the connection reads a chunk of data.
+     *
+     * @param size size of the data chunk read
+     */
+    void dataRead(int size);
```

Comment:
The method names here are in past tense "dataWritten", "flushed", etc. Is this intentional that they are associated with timing? While invoking these callbacks we are not consistent; `dataRead()` is after we have read from socket, `dataWritten()` is before we have written to the socket. 

An alternate is to use `onDataRead`, `onDataWrite`, `onFlush` ... etc or `whenDataRead`, `whenFlushed` which hides the timing (before/after) and are also similar to the callbacks on our async primitives.